pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -273,9 +274,27 @@ func (i *InfraProvider) Provision(ctx context.Context, dir string, parents asset
 	} else {
 		logrus.Debugf("No Ignition requirements for the %s provider", i.impl.Name())
 	}
-	bootstrapIgnSecret := IgnitionSecret(bootstrapIgnData, clusterID.InfraID, "bootstrap")
-	masterIgnSecret := IgnitionSecret(masterIgnAsset.Files()[0].Data, clusterID.InfraID, "master")
-	machineManifests = append(machineManifests, bootstrapIgnSecret, masterIgnSecret)
+
+	switch i.impl.Name() {
+	case nutanixtypes.Name:
```

Comment:
It seems that the "IgnitionProvider" interface is used to customize the bootstrap ignition data only. And here for the Nutanix use case, we want to generate the ignition data for each control-plane machine separately to create the corresponding secret manifests. So I think it's simpler to handle the Nutanix special case here instead of modify the "IgnitionProvider" interface and the "IgnitionInput" struct, for they are used by other platform providers.

Reply:
The contents of ignition (bootstrap & master) are generated during asset generation--the IgnitionProvider should not change the contents of the bootstrap ignition; rather it handles how the ignition file is stored in the cloud and provided to the bootstrap node. 

So it seems these changes should go into [the master ignition asset](https://github.com/openshift/installer/blob/master/pkg/asset/ignition/machine/master.go), 

possibly here:
https://github.com/openshift/installer/blob/master/pkg/asset/ignition/machine/node.go#L27

If you make the changes as in this PR, you won't see the hostnames in the ignition configs when you run `openshift-install create ignition-configs`, but if you move your changes to the ignition asset, you should see your changes reflected in those files.

Reply:
@patrickdillon Thanks for the comments. I will change the code based on your recommendation.

Reply:
> But it seems not work for the master ignition data, since the machine name for machine names of the master machines are different.
@yanhua121 can you explain what you mean by this?

Reply:
Oh it's because all master machines use the same ignition file but each master has its own name?

Reply:
Was this not a problem with terraform-based provisioning?

Reply:
@r4f4 This problem is handled the same way in the terraform provisioning, see https://github.com/openshift/installer/blob/master/data/data/nutanix/cluster/main.tf#L53-L63.

Reply:
Oh I see, so ignition files are generated per master... I suppose we could extend the ignition interface to allow overriding master ignition. I'm willing to enable that to unblock the regression, and will take a closer look at doing that tomorrow; but I wonder if this isn't a bug in RHCOS or the MCO? I don't think the installer (and the machine actuator) should be setting the hostname like this.